server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/sse/S3SSECConfiguration.java

```diff
@@ -23,20 +23,23 @@
 
 import com.google.common.base.Preconditions;
 
-public record S3SSECustomerConfiguration(boolean enabled,
-                                         String algorithm,
-                                         String masterPassword,
-                                         String salt) {
+public interface S3SSECConfiguration {
 
-    public static final List<String> SUPPORTED_ALGORITHMS = List.of("AES256");
+    List<String> SUPPORTED_ALGORITHMS = List.of("AES256");
 
-    public S3SSECustomerConfiguration(boolean enabled, String algorithm, String masterPassword, String salt) {
-        this.enabled = enabled;
-        if (enabled) {
-            Preconditions.checkArgument(SUPPORTED_ALGORITHMS.contains(algorithm), "Unsupported algorithm: %s. The supported algorithms are: %s", algorithm, SUPPORTED_ALGORITHMS);
+    String algorithm();
+
+    record Basic(String algorithm,
+                 String masterPassword,
+                 String salt) implements S3SSECConfiguration {
+        public Basic {
+            Preconditions.checkArgument(SUPPORTED_ALGORITHMS.contains(algorithm), "Unsupported algorithm: " + algorithm + ". The supported algorithms are: " + SUPPORTED_ALGORITHMS);
+        }
+    }
+
+    record KeyDerivationEnabled(String algorithm) implements S3SSECConfiguration {
```

Comment:
Out of scope for this task?

Reply:
No answer to that remark?

Reply:
Indeed, this remark is completely correct 100%. 
I'll remove it in the next fixup commit.